pkg/event/event.go

```diff
@@ -36,14 +38,16 @@ import (
 )
 
 type event struct {
-	metadata map[string]interface{}
-	payload  interface{}
-	ctx      context.Context
-	ack      ack.SubTree
-	tid      tenant.Id
-	spanName string
-	span     trace.Span //Only valid in the root event
-	created  time.Time
+	metadata           map[string]interface{}
+	payload            interface{}
+	ctx                context.Context
+	ack                ack.SubTree
+	tid                tenant.Id
+	spanName           string
+	span               trace.Span //Only valid in the root event
+	tracePayloadOnNack bool
```

Comment:
Do we actually need this variable? Or can we infer it from whether tracePayload == nil or not?

Reply:
we could infer from tracePayload != nil but i'm not sure this makes it more readable